charts/hono/templates/prometheus/prometheus-config.yaml

```diff
@@ -14,7 +14,7 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  {{- $args := dict "dot" . "component" "metrics" "name" "prometheus-config" }}
+  {{- $args := dict "dot" (. | deepCopy | merge (dict "Values" (pick .Values.prometheus.server "nameOverride" "fullnameOverride"))) "component" "metrics" "name" "prometheus-config" }}
```

Comment:
The Prometheus chart expects the ConfigMap name to be `{ReleaseName}-{.Values.prometheus.server.configMapOverrideName}`, i.e. `{ReleaseName}-prometheus-config`.
But the code above doesn't produce that with the current values.yaml.

I would suggest using the following:
```
  {{- /* prometheus chart expects metadata.name to be "{ReleaseName}-{prometheus.server.configMapOverrideName}" */}}
  {{- $prometheusDotScope := . | deepCopy | merge (dict "Values" (dict "fullnameOverride" .Release.Name)) }}
  {{- $args := dict "dot" $prometheusDotScope "component" "metrics" "name" .Values.prometheus.server.configMapOverrideName }}
```